src/applications/vaos/new-appointment/components/CommunityCareProviderSelectionPage/index.jsx

```diff
@@ -0,0 +1,154 @@
+import React, { useEffect } from 'react';
+import { useHistory } from 'react-router-dom';
+import { connect } from 'react-redux';
+import SchemaForm from 'platform/forms-system/src/js/components/SchemaForm';
+import FormButtons from '../../../components/FormButtons';
+import { LANGUAGES } from '../../../utils/constants';
+import * as actions from '../../redux/actions';
+import { getFormPageInfo } from '../../../utils/selectors';
+import { scrollAndFocus } from '../../../utils/scrollAndFocus';
+import ProviderSelectionField from './ProviderSelectionField';
+
+const initialSchema = {
+  type: 'object',
+  required: ['preferredLanguage', 'communityCareProvider'],
+  properties: {
+    communityCareSystemId: {
+      type: 'string',
+      enum: [],
+    },
+    preferredLanguage: {
+      type: 'string',
+      enum: LANGUAGES.map(l => l.id),
+      enumNames: LANGUAGES.map(l => l.text),
+    },
+    communityCareProvider: {
+      type: 'object',
+      properties: {
+        noProvider: {
+          type: 'boolean',
+        },
+        providerData: {
+          type: 'object',
+          properties: {
+            uniqueId: {
+              type: 'string',
+            },
+            name: {
+              type: 'string',
+            },
+            phone: {
+              type: 'string',
+            },
+            lat: {
+              type: 'number',
+            },
+            long: {
+              type: 'number',
+            },
+            address: {
+              type: 'object',
+              properties: {
+                street: {
+                  type: 'string',
+                },
+                city: {
+                  type: 'string',
+                },
+                state: {
+                  type: 'string',
+                },
+                zip: {
+                  type: 'string',
+                },
+              },
+            },
+          },
+        },
+      },
+    },
+  },
+};
+
+const uiSchema = {
+  communityCareSystemId: {
+    'ui:title': 'What’s the closest city and state to you?',
+    'ui:widget': 'radio',
+  },
+  preferredLanguage: {
+    'ui:title':
+      'Do you prefer that your community care provider speak a certain language?',
+  },
+  communityCareProvider: {
+    'ui:options': {
+      showFieldLabel: true,
+    },
+    'ui:field': ProviderSelectionField,
```

Comment:
Custom field can be hooked up here and it'll control the checkbox as well as the list stuff.